code/script.js

```diff
@@ -29,23 +37,246 @@ const showMessage = (message, sender) => {
       </section>
     `
   }
-  // This little thing makes the chat scroll to the last message when there are too many to be shown in the chat box
   chat.scrollTop = chat.scrollHeight;
 }
 
-// Starts here
-const greetUser = () => {
-  // here we call the function showMessage, that we declared earlier with the argument "Hello there, What's your name?" for message, and the argument "bot" for sender
-  showMessage("Hello there, What's your name?", 'bot');
-  // Just to check it out, change 'bot' to 'user' here 👆
+//Simplifies calling for user and bot reply in the chat bubbles, adding a delay for both types.
+const botReply = (msg) => {
+  setTimeout(showMessage, 1000, msg, 'bot');
+}
+const userReply = (msg) => {
+  setTimeout(showMessage, 300, msg, 'user');
+}
+
+
+const greetUser = (timeGreetings, hour) => {
+
+if (hour >= 5 && hour <= 10) {
+    botReply(timeGreetings[0]);           //morning (dvs 5 till 10)
+} else if (hour > 10 && hour <= 18) {
+    botReply(timeGreetings[1]);           //hello (dvs 11 till 18)
+} else if (hour > 18 && hour <= 22) {
+    botReply(timeGreetings[2]);           //evening (dvs 19 till 22)
+} else {
+    botReply(timeGreetings[3]);           //late (dvs 23 till 4)
+    botReply("Since you're already up, What's your name gorgeous?");
+}
+};
+
+
+//After user has submitted their name, this runs:
+form.addEventListener('submit', (event) => {
+  event.preventDefault()
+  
+  let firstName = document.getElementById('firstName').value;
+  userReply(firstName);
+  recognizeName(firstName);
+})
+
+
+
+const recognizeName = (firstName) => {
+
+  botReply(`How can I help you, ${firstName}?`);
+
+  inputWrapper.innerHTML =
+      `<button id="appointmentBtn" type="submit" value="Book appointment">Book appointment</button>
+    <button id="openingHoursBtn" type="submit" value="Opening hours">Opening hours</button>
+    <button id="styleAdviceBtn" type="submit" value="Style advice">Style advice</button>`;
+
+  
+  let appointmentBtn = document.getElementById('appointmentBtn');
```

Comment:
Would it be cleaner to make an eventListener for the three of them using an if-statement?